### PR TITLE
Adding Percent to DataGen

### PR DESCRIPTION
**Pull request by blaynem**
Adds the data provider for a PercentFormatter. Solving the data portion of #4483

### CODE REVIEW

**Comment by Manishearth** [comment]
```
Test data needs to be updated. Please run `cargo make download-repo-sources`, `cargo make testdata` and `cargo make testdata-hello-world:

 M provider/datagen/tests/data/postcard/fingerprints.csv
?? provider/datagen/tests/data/json/percent/
```

Still shows up in fingerprints even if it's test-only, I think.

**Review by Manishearth — COMMENTED** [review]
overall this looks good. I'm not super familiar with this module so I'd defer to @robertbastian or @younies

File: provider/datagen/src/transform/cldr/percent/mod.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+fn extract_percent_essentials<'data>(
+    numbers_resource: &cldr_serde::numbers::Resource,
+) -> Result<PercentEssentialsV1<'data>, DataError> {
+    let percent_formats = &&numbers_resource
+        .main
+        .value
+        .numbers
+        .numsys_data
+        .percent_patterns
+        .get(&tinystr!(8, "latn"))
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/percent/mod.rs:54]: Note: We'll want to support multiple numbering systems; see how we do it in DecimalSymbols. Please add a TODO.

File: provider/datagen/src/transform/cldr/percent/mod.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        .ok_or_else(|| DataError::custom("Could not find the standard pattern"))?;
+
+    let standard = &percent_formats.standard;
+
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/percent/mod.rs:58]: Somewhere in here you'll want to do the following:

1. Remove the number placeholder, like `#,##0`, from the string, and keep track of its index in the string (will often but not necesarilly be zero)
2. Replace the `%` character with the localized percent symbol, which you should be able to pull also from the CLDR JSON data file

**Review by sffc — DISMISSED** [review]
Good start!

File: provider/datagen/src/transform/cldr/percent/mod.rs (hunk 1 above)
Comment (blaynem) [reply provider/datagen/src/transform/cldr/percent/mod.rs:55]: Added some TODOs on top like how it was in Currency. I checked through a few of the different locales in [CLDR data](https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-numbers-full/main/el/numbers.json#L94) and only ever saw the one `percentFormats-numberSystem-latn` property.

Is just the TODO good for now? Or am I looking in the wrong spot for those other numbering systems?

File: provider/datagen/src/transform/cldr/percent/mod.rs (hunk 2 above)
Comment (blaynem) [reply provider/datagen/src/transform/cldr/percent/mod.rs:94]: Updated!

File: experimental/dimension/src/provider.rs [hunk 3]
```diff
@@ -114,3 +117,55 @@ pub struct CurrencyPatterns {
...
+#[derive(Default, Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_dimension::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct PercentEssentialsV1<'data> {
+    /// The index of the number placeholder in the standard pattern.
+    pub number_index: u16,
```
Comment (robertbastian) [inline experimental/dimension/src/provider.rs:131]: do we expect patterns longer than 256?

File: provider/datagen/tests/data/json/percent/essentials@1/und.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "number_index": 0,
+  "percent_sign_affixes": {
+    "prefix": "",
+    "suffix": ""
+  },
+  "percent_sign_symbol": "%",
+  "percent_symbol_index": 5,
+  "standard": "#,##0%"
```
Comment (robertbastian) [inline provider/datagen/tests/data/json/percent/essentials@1/und.json:9]: is `standard` fully determined by the locales number format, and the other fields?

Comment (blaynem) [reply provider/datagen/tests/data/json/percent/essentials@1/und.json:9]: `standard` is just a pass through from the actual object
```
"percentFormats-numberSystem-latn": {
    "standard": "#,##0%"
},
```

File: experimental/dimension/src/provider.rs (hunk 3 above)
Comment (blaynem) [reply experimental/dimension/src/provider.rs:131]: Likely never, updated to u8. Good catch :)

File: provider/datagen/src/transform/cldr/percent/mod.rs (hunk 1 above)
Comment (sffc) [reply provider/datagen/src/transform/cldr/percent/mod.rs:55]: TODOs are fine.

**Review by sffc — APPROVED** [review]
Looks good as a base for merging to experimental and then we can build it out and fix the TODOs

**Pull request merged**